# Failure of sending

fun goAlright() {
  var ap = new(); # Yay synchronisation!
  try {
    var s = fork (fun (s) {
  #    cancel(s);
      ignore(request(ap));
      ignore(receive(s));
    });
    ignore(accept(ap));
    ignore(send(5, s))
  } as (_) in {
    "send successful"
  } otherwise {
    "exception"
  }
}


fun go() {
  var ap = new(); # Yay synchronisation!
  try {
    var s = fork (fun (s) {
      cancel(s);
      ignore(request(ap))
    });
    ignore(accept(ap));
    ignore(send(5, s))
  } as (_) in {
    "send successful"
  } otherwise {
    "exception"
  }
}

go()
#goBadly()
#goAlright()
